application/src/Api/Representation/AbstractResourceEntityRepresentation.php

```diff
@@ -501,11 +501,28 @@ public function displaySubjectValues($page = null, $perPage = null, $property =
      * Get the display title for this resource.
      *
      * @param string|null $default
+     * @param string|null $lang
      * @return string|null
      */
-    public function displayTitle($default = null)
+    public function displayTitle($default = null, $lang = null)
     {
-        $title = $this->title();
+        $template = $this->resourceTemplate();
+        if ($template && $template->titleProperty()) {
+            $titleTerm = $template->titleProperty()->term();
+        } else {
+            $titleTerm = 'dcterms:title';
+        }
+
+        if ($lang !== null) {
+            if ($titleValue = $this->value($titleTerm, ['lang' => $lang])) {
+                $title = $titleValue->value();
+            }
+        }
+
+        if (!isset($title) || $title === null) {
```

Comment:
I'd prefer not to have these "is not set or is null" checks: it's cleaner probably to just define `$title` as null near the top and then not need the isset check.

Reply:
OK updated in 81ec357